modules/affile/logproto-file-writer.c

```diff
@@ -134,6 +137,26 @@ log_proto_file_writer_flush(LogProtoClient *s)
   self->buf_count = 0;
   self->sum_len = 0;
 
+  // emit a "flush" signal to the transport, so that it can react to the error
+  GString *request_body = g_string_new("");
+  g_string_append_printf(request_body, "%d", self->buf_size);
+  GString *file_name = g_string_new("");
+  g_string_append_printf(file_name, "%s", self->super.transport->name);
+  FileFlushSignalData signal_data =
+  {
+    .filename = file_name,
+    .size = self->buf_size, // implementation yet to be decided
+    .interval = 0, // implementation yet to be decided
+  };
+
+  //self->signal_slot_connector = self->signal_slot_connector_driver->super.super.super.signal_slot_connector;
+  //The flush-signal is not emitted on the AFFileDestDriver's signal-slot-connector, it is emitted using a completely different connector.
+  //The reason for this is that the AFFileDestDriver is not a LogProtoClient, but a LogProtoDestDriver.
+
+  //Register the signal to AFFileDestDriver's signal-slot-connector (the signal is emitted on the AFFileDestDriver's signal-slot-connector)
+
+  EMIT(self->signal_slot_connector, signal_file_flush, &signal_data);
+
```

Comment:
Please remove all the comments here. We can get rid of the test data as well for now.
We can emit the signal with an empty `FileFlushSignalData`. The next task (and PR) will be about filling this data properly.

Reply:
:+1: